packages/mobile/src/geth/GethNativeBridgeSigner.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  encodeTransaction,
+  extractSignature,
+  RLPEncodedTx,
+  rlpEncodedTx,
+} from '@celo/contractkit/lib/utils/signing-utils'
+import { Signer } from '@celo/contractkit/lib/wallets/signers/signer'
+import { ensureLeading0x } from '@celo/utils/lib/address'
+import { normalizeAddressWith0x } from '@celo/utils/src/address'
+import * as ethUtil from 'ethereumjs-util'
+import { GethNativeModule } from 'react-native-geth'
+import Logger from 'src/utils/Logger'
+import { Tx } from 'web3-core'
+
+const INCORRECT_PASSWORD_ERROR = 'could not decrypt key with given password'
+const currentTimeInSeconds = () => Math.floor(Date.now() / 1000)
+
+const TAG = 'geth/GethNativeBridgeSigner'
+/**
+ * Implements the signer interface on top of a RNGeth instance
+ */
+export class GethNativeBridgeSigner implements Signer {
+  /**
+   * Construct a new instance of the RPC signer
+   *
+   * @param geth the RNGeth instance
+   * @param account Account address derived from the private key to be called in init
+   * @param unlockBufferSeconds Number of seconds to shrink the unlocked duration by to account for
+   * latency and timing inconsistencies on the node
+   * @param unlockTime Timestamp in seconds when the signer was last unlocked
+   * @param unlockDuration Number of seconds that the signer was last unlocked for
+   */
+  constructor(
+    protected geth: GethNativeModule,
+    protected account: string,
+    protected unlockBufferSeconds = 5,
+    protected unlockTime = -1,
```

Comment:
Most of these are copied from the `RpcSigner` because I wanted to mimic its functionality, but I agree it's better to be explicit so I'll fix these here and in the `RpcSinger` as well. It might make sense to actually abstract all of this in a super class that both signers extend to DRY things up a bit, but it might also intermingle things more, I have a rule of thumb to only start DRYing things after the 3rd repetition :D